cpp/test/sg/fil_test.cu

```diff
@@ -283,8 +392,49 @@ std::vector<FilTestParams> inputs = {
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f},
 };
 
-TEST_P(FilTest, Predict) { compare(); }
+TEST_P(PredictFilTest, Predict) { compare(); }
+
+INSTANTIATE_TEST_CASE_P(FilTests, PredictFilTest,
+                        testing::ValuesIn(predict_inputs));
+
+std::vector<FilTestParams> import_inputs = {
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0, fil::algo_t::NAIVE, 42,
```

Comment:
These are a nice mix of tests but it would be great if we could also test some real-world trees... Since these trees are all generated by our random generation code, it seems like we don't have particularly weird cases. A couple of thoughts (should probably be a separate pull request):

- treelite appears to have a couple of example trees (stored as prototxt)... could we import those and confirm we match their inference?

- What about confirming that we match xgboost's predictions? If we have a saved xgboost model, we could just save its predictions and check against that.

- Also, it'd be interesting to see a test for the "input tree is 'sparse' so we convert to dense" case... I think that is handled in the code by just leaving blank nodes in the node array, but it's always nice to test...

Any other thoughts on additional tests? It seems really easy to end up with slight deviations from the input tree that will drive customers crazy, so it's probably worth putting additional thought into it. This is probably easier to do at the python layer though.